compiler/frontend/src/mir/expression.rs

```diff
@@ -27,6 +27,7 @@ pub enum Expression {
     /// is fuzzable or not, this parameter may be used to dynamically determine
     /// who's at fault if some `needs` is not fulfilled.
     Lambda {
+        original_hirs: Vec<hir::Id>,
```

Comment:
```suggestion
        original_hirs: FxHashSet<hir::Id>,
```

Makes more sense semantically, but probably not necessary